dvc/commands/repro.py

```diff
@@ -156,13 +156,10 @@ def add_parser(subparsers, parent_parser):
     add_arguments(repro_parser)
     # repro only args
     repro_parser.add_argument(
```

Comment:
@dberenbaum @daavoo to really match the order in https://github.com/iterative/dvc.org/pull/3345/files#diff-6c1f3192f09e2722ba169e9fa219b3b5158bbafa470b382c2d6135db7aa1e20d we'd need to change this logic, probably, affecting `exp run` as well. Should I go ahead with that?

Reply:
The alternative is to keep this order due to necessity and change it again in docs.

Reply:
> to really match the order [we agreed on](https://github.com/iterative/dvc.org/pull/3345/files#diff-6c1f3192f09e2722ba169e9fa219b3b5158bbafa470b382c2d6135db7aa1e20d) we'd need to change this logic

With that I meant that the `add_arguments(repro_parser)` function call groups a bunch of options and then a few others are added after that call so without changing or flattening that fn. in the current scope we can't match the agreed order.

> we should probably reorder it so that --glob comes later ... my suggestion would be this order

I like the [current order](https://dvc.org/doc/command-reference/repro#synopsis) more but up to you. I also don't think a bug should define the help output order or docs (but `--glob` could be hidden in `exp run` for now if needed).

> > having to change exp run code for consistency 
>
> we should also have updated exp run in https://github.com/iterative/dvc.org/pull/3345 for consistency

Also, I'm not sure about this any more! The `exp run` ref [already has](https://dvc.org/doc/command-reference/exp/run#synopsis) a `{ remote options ... }` line at the moment and all the other options are unique I think... Do we mean reorder those other options? I'll open a separate discussion on reordering `exp run` not to overcomplicate this review.

Reply:
> > to really match the order [we agreed on](https://github.com/iterative/dvc.org/pull/3345/files#diff-6c1f3192f09e2722ba169e9fa219b3b5158bbafa470b382c2d6135db7aa1e20d) we'd need to change this logic
> 
> With that I meant that the `add_arguments(repro_parser)` function call groups a bunch of options and then a few others are added after that call so without changing or flattening that fn. in the current scope we can't match the agreed order. And I'm not 100$ sure why that function exists so I'm hesitant to touch it.

Sorry, I got what you meant but didn't explain my reasoning. That function exists to be reused in `exp run`, and I don't think we should break it up. Putting the common options (between `repro` and `exp run`) first not only simplifies the code but also provides some logical order and makes the help text more consistent. To do that, `--no-commit` and `--no-run-cache` need to be at the end (and technically `--glob`, see below). Given those constraints, is there an order you prefer?

> > we should probably reorder it so that --glob comes later ... my suggestion would be this order
> 
> I like the [current order](https://dvc.org/doc/command-reference/repro#synopsis) more but up to you. I also don't think a bug should define the help output or docs (but `--glob` could be hidden in `exp run` for now if needed).

True. Dealing with `--glob` might get a bit messy then in the code, but if you there's a good reason to put it in the middle of the other options, we can work around it.

> > > having to change exp run code for consistency
> > 
> > 
> > we should also have updated exp run in [iterative/dvc.org#3345](https://github.com/iterative/dvc.org/pull/3345) for consistency
> 
> Also, I'm not sure about this any more! The `exp run` ref [already has](https://dvc.org/doc/command-reference/exp/run#synopsis) a `{ remote options ... }` line at the moment i.e. it's consistent with `repro`, and all the other options are unique I think... Do we mean reorder those other options? I'll open a separate discussion on reordering `exp run` not to overcomplicate this review.

Yeah, I missed how we handle that in the docs. I think it's fine as is.

Reply:
> That function exists to be reused in exp run, and I don't think we should break it up. Putting the common options (between repro and exp run) first not only simplifies the code but also provides some logical order and makes the help text more consistent.

Got it, OK. And I see that `--glob` is not in it as of now so fine, let's leave the 3 repro-only ones ([mentioned](https://dvc.org/doc/command-reference/exp/run#options) in the `exp run` ref.) at the end then, but not sure about the other changes in your suggestion. Maybe

```ini
usage: dvc repro [-h] [-q | -v] [-f] [-i]
                 [-s] [-p] [-P] [-R] [-m]
                 [--downstream] [--force-downstream]
                 [--pull] [--dry]  # --glob can be moved up here later.
                 [--glob] [--no-commit] [--no-run-cache]
                 [targets [<target> ...]]
```

> Will suggest in https://github.com/iterative/dvc.org/pull/3435 too.